lib/net/http_fetch_plugin.js

```diff
@@ -130,6 +130,9 @@ shaka.net.HttpFetchPlugin = class {
       const contentLengthRaw = response.headers.get('Content-Length');
       const contentLength =
           contentLengthRaw ? parseInt(contentLengthRaw, 10) : 0;
+      if (contentLength > 0) {
+        progressUpdated(0, 0, contentLength, contentLength);
+      }
```

Comment:
Could you please explain why you call progressUpdated() here?

Reply:
to notify total segment size as soon as we have headers

do you think that's a premature optimization?

Reply:
It seems reasonable, but perhaps you should add a comment explaining why you do this.